keras/backend/tensorflow_backend.py

```diff
@@ -859,7 +859,8 @@ def set_value(x, value):
     '''Sets the value of a tensor variable,
     from a Numpy array.
     '''
-    tf_dtype = _convert_string_dtype(dtype(value))
+    value = np.asarray(value)
+    tf_dtype = _convert_string_dtype((x*1.).dtype)
```

Comment:
Use `x.dtype.name.split('_')[0]`, it avoids creating a new operation in the graph.

Have you checked that your fix resolves the graph size increase issue?